ompi/proc/proc.h

```diff
@@ -369,16 +369,42 @@ static inline bool ompi_proc_is_sentinel (ompi_proc_t *proc)
     return (intptr_t) proc & 0x1;
 }
 
-static inline intptr_t ompi_proc_name_to_sentinel (opal_process_name_t name)
+/*
+ * we assume an ompi_proc_t is at least aligned on two bytes,
+ * so if the LSB of a pointer to an ompi_proc_t is 1, we have to handle
+ * this pointer as a sentinel instead of a pointer.
+ * a sentinel can be seen as a 64 bits array with the following format :
+ * - bit  0     : 1
+ * - bits 1-15  : local jobid
+ * - bits 16-31 : job family
+ * - bits 32-63 : vpid
```

Comment:
I agree with using the MSB of the local jobid. What I'm confused by is that your documented bitmap is incorrect. The local jobid is in the middle of the object, not at the edge as you show.

Nathan mistakenly used the MSB of the entire structure instead of the local jobid. Your bitmap moves it to the LSB of the entire structure, which is the LSB of the vpid, not the local jobid.